source/data-modeling/indexes.txt

```diff
@@ -0,0 +1,219 @@
+.. _mongoid-work-with-indexes:
+
+=================
+Work With Indexes
+=================
+
+.. default-domain:: mongodb
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, odm, optimization, efficiency, Atlas search
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 3
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use **indexes** with {+odm+}. Indexes can
+improve the efficiency of queries by limiting the number of documents MongoDB
+needs to scan. If your application is repeatedly running queries
+on certain fields, you can create an index on those fields to improve query
+performance.
+
+The following sections in this guide describe how to declare and create different
+types of indexes using {+odm+}. The examples use the ``Restaurant`` model, which 
+maps to the ``restaurants`` collection in the ``sample_restaurants`` database. 
+To learn how to connect to this database
+and collection using {+odm+}, see the :ref:`mongoid-quick-start-rails` or 
+:ref:`mongoid-quick-start-sinatra` guides.
+
+Declare and Create an Index
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using {+odm+}, you can declare your index using the ``index`` macro and 
+then create it using the ``create_indexes`` macro.
+
+The following code example shows how to declare and create an ascending index 
+named ``cuisine_index`` on the ``cuisine`` field in the ``Restaurant`` class:
+
+.. literalinclude:: /includes/data-modeling/indexes.rb
+  :language: ruby
+  :emphasize-lines: 8, 11
+  :start-after: start create index
+  :end-before: end create index
+
+The ``index`` macro defines the index you want to create and the ``create_indexes``
+macro creates it in the ``restaurants`` collection. 
+
+When defining an index, the first hash object contains the field you want to
+index and its direction. ``1`` represents an ascending index, and ``-1`` represents a
+descending index. The second hash object contains index options. To learn more
+about index options, see the :ref:`mongoid-indexes-api-documentation` section.
+
+Aliases and Declaring Indexes
++++++++++++++++++++++++++++++
+
+You can use aliased field names in index definitions. For example, the following
+code creates a field on the ``b`` field, which is an alias of the ``borough``
+field:
```

Comment:
[q] Should this be index? I wasn't sure on initial read.
```suggestion
code creates an index on the ``b`` field, which is an alias of the ``borough``
field:
```

Reply:
yes! thank you so much for catching